spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -38,8 +38,7 @@ private ConfigUtils() {
 	}
 
 	public static String getApplicationName(Environment env, String configName, String configurationTarget) {
-		if (StringUtils.isEmpty(configName)) {
-			// TODO: use relaxed binding
+		if (!StringUtils.hasLength(configName)) {
```

Comment:
first, use `hasLength` and second remove the comment about relaxed binding, as I think this is already taken care of internally, when `SystemEnvironmentPropertySource`, for example, is going to get hit.